src/dmd/expressionsem.d

```diff
@@ -10946,6 +10947,16 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             result = e2x;
             return;
         }
+        if (e2x.type.ty == Tnoreturn)
+        {
+            // rewrite as (e1x && e2x),false or (e1x || e2x),true
+            exp.type = Type.tvoid;
+            exp.e1 = e1x;
+            exp.e2 = e2x;
+            auto bres = new IntegerExp(Loc.initial, exp.op == TOK.orOr, Type.tbool);
```

Comment:
```suggestion
            auto bres = new IntegerExp(exp.loc, exp.op == TOK.orOr, Type.tbool);
```

Reply:
There's really no need since that node will get optimized away.

Reply:
Er, ignore that last comment.